core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -64,12 +65,19 @@ oppia.directive('audioControls', [
           };
 
           $scope.playPauseAudioTranslation = function() {
-            // TODO(tjiang11): Change from on-demand loading to pre-loading.
-
-            // TODO(tjiang11): On first play, ask learner to pick language
-            // and subsequently for confirmation to use bandwidth 
-            // to download audio files.
+            if (!AudioPreloaderService.hasPreloaded()) {
+              if (getCurrentAudioTranslation()) {
```

Comment:
In what circumstance does this condition evaluate to true? I get this popup modal as soon as I try to play any one file...

Also, I have a general question about this. The options available to me as a learner seem to be preload-or-nothing, and the former option downloads all the audio in the entire exploration. Do I not get the option to just download the specific audio files I need on an as-needed basis? (I looked at the doc, but I'm not sure that this part was written out in detail...or maybe I misunderstood it.)